lib/spack/spack/test/concretize.py

```diff
@@ -525,3 +526,38 @@ def test_regression_issue_7941(self):
         t.concretize()
 
         assert s.dag_hash() == t.dag_hash()
+
+    @pytest.mark.parametrize('abstract_specs', [
+        # Establish a baseline - concretize a single spec
+        ('mpileaks',),
+        # When concretized together with older version of callpath
+        # and dyninst it uses those older versions
+        ('mpileaks', 'callpath@0.9', 'dyninst@8.1.1'),
+        # Handle recursive syntax within specs
+        ('mpileaks', 'callpath@0.9 ^dyninst@8.1.1', 'dyninst'),
```

Comment:
Can you add a case with two packages that share some dependencies but neither is a dependency of the other? Sorry that I didn't make it clearer in my review, I think that's a case we're missing since dyninst and callpath are both dependencies of mpileaks.

I think `mpileaks` and `direct-mpich` would be a good choice of overlapping packages from the mock repo.

Reply:
Done